lib/vm.js

```diff
@@ -79,20 +79,34 @@ Script.prototype.runInNewContext = function(sandbox, options) {
   return this.runInContext(context, options);
 };
 
-function validateString(prop, propName) {
-  if (prop !== undefined && typeof prop !== 'string')
+function validate(type, prop, propName) {
+  if (prop !== undefined &&
+      ((type === 'Object' && prop === null) ||
+      typeof prop !== type.toLowerCase()))
```

Comment:
This logic is overly complicated, and would disallow V8 from inlining the typeof check. How it was (one validation function per type) was just fine.